nodegsettings.cc

```diff
@@ -0,0 +1,385 @@
+		return scope.Close(Undefined());
+	}
+	GSettingsSchema * gsettings_schema =
+		g_settings_schema_source_lookup (schema_source,
+		                                 schemaId,
+		                                 FALSE);
+	if ( ! schema_source ) {
+		printf("Schema '%s' is not installed!\n", schemaId);
+		ThrowException(Exception::Error(String::New("Schema is not installed!")));
+		return scope.Close(Undefined());
+	}
+	gboolean has_key = g_settings_schema_has_key (gsettings_schema, key);
+	if ( ! has_key ) {
+		printf("Key '%s' does not exist!\n", key);
+		ThrowException(Exception::Error(String::New("Key does not exist!")));
+		return scope.Close(Undefined());
+	}
+	GSettingsSchemaKey * gsettings_key =
+		g_settings_schema_get_key (gsettings_schema, key);
+
+	// get required variant type
+	settings = g_settings_new(schemaId);
+	variant  = g_settings_get_value(settings,key);
+	type     = g_variant_get_type(variant);
+
+
+	if ( g_variant_type_equal(type,G_VARIANT_TYPE_BOOLEAN) ) {
+		if ( value_obj->IsBoolean() ) {
+			success = g_settings_set_boolean(settings,key,value_obj->BooleanValue());
+		} else {
+			ThrowException(Exception::Error(String::New("Key requires boolean value!")));
+			return scope.Close(Undefined());
+		}
+	}
+	else if ( g_variant_type_equal(type,G_VARIANT_TYPE_STRING) ) {
+		if ( value_obj->IsString() ) {
+			char *val = v8_value_to_utf8_string(value_obj);
+
+			success = g_settings_set_string(settings,key,val);
+		} else {
+			ThrowException(Exception::Error(String::New("Key requires string value!")));
+			return scope.Close(Undefined());
+		}
+	}
+	else if ( g_variant_type_equal(type, G_VARIANT_TYPE_DOUBLE) ) {
+		if ( value_obj->IsNumber() ) {
+			gdouble v8_double_value = value_obj->ToNumber()->Value();
+			GVariant *variant_to_set = g_variant_new_double(v8_double_value);
+
+			gboolean is_valid = g_settings_schema_key_range_check (gsettings_key, variant_to_set);
+			if ( ! is_valid ) {
+				printf("Invalid range or type of '%s'\n", key);
+				ThrowException(Exception::Error(String::New("Invalid range or type!")));
+				return scope.Close(Undefined());
+			}
+
+			success = g_settings_set_value(settings, key, variant_to_set);
+		} else {
+			ThrowException(Exception::Error(String::New("Key requires a number!")));
+			return scope.Close(Undefined());
+		}
+	}
+	else if ( g_variant_type_equal(type, G_VARIANT_TYPE_INT32) ) {
+		if ( value_obj->IsInt32() ) {
+			GVariant *variant_to_set = g_variant_new_int32 (value_obj->ToInt32()->Value());
+
+			gboolean is_valid = g_settings_schema_key_range_check (gsettings_key, variant_to_set);
+
+			if ( ! is_valid ) {
+				printf("Invalid range or type of '%s'\n", key);
+				ThrowException(Exception::Error(String::New("Invalid range or type!")));
+				return scope.Close(Undefined());
+			}
+
+			success = g_settings_set_value(settings, key, variant_to_set);
+		} else {
+			ThrowException(Exception::Error(String::New("Key requires a integer number!")));
+			return scope.Close(Undefined());
```

Comment:
Is this code reachable? I'm not knowledgeable about V8 exception handling patterns. If this followed C++ semantics, it would have to go into a finally block.